Separate concerns of `Bundler::SharedHelpers#set_bundle_environment`

RochesterinNYC opened the pull request:
- Splits the implementation details/sections of `Bundler::SharedHelpers#set_bundle_environment` into three more modular, singularly responsible methods: `set_path`, `set_rubyopt`, `set_rubylib`
- Adds unit tests for `set_bundle_environment` and the new methods

RochesterinNYC commented:
Not 100% sure whether `set_path`, `set_rubyopt`, `set_rubylib` should be private or public. Made them public as of right now since `set_bundle_environment` was essentially a wrapper around the three.

segiddins commented:
Private imho

RochesterinNYC commented:
Hm, okay, so then no unit tests for those methods? 

And then we would keep the `set_bundle_environment` mocked call unit spec or create/make a unit test that aggregates all the functional unit testing for the set methods?

segiddins commented:
@RochesterinNYC yup!

segiddins commented:
Some rubocop offenses in the specs: https://travis-ci.org/bundler/bundler/jobs/101209168#L264-L277

RochesterinNYC commented:
Updated the PR with making the `set_` methods private. Also added `ENV['RUBYLIB']` and `ENV['RUBYOPT']` to the after spec reset list to prevent the new `set_bundle_environment` unit tests from having side effects.

segiddins commented:
@homu r+

homu commented:
:pushpin: Commit 0a1c1ca has been approved by `segiddins`

<!-- @homu r=segiddins 0a1c1cac447aba9eaf101823de57bb4d1b9d73f4 -->

homu commented:
:zap: Test exempted - [status](https://travis-ci.org/bundler/bundler/builds/101321811)
